packages/aws-cdk/test/util.ts

```diff
@@ -49,11 +49,31 @@ function clone(obj: any) {
   return JSON.parse(JSON.stringify(obj));
 }
 
+function addNestedStacks(templatePath: string, outdir: string, rootStackTemplate?: any) {
+  let template = rootStackTemplate;
+
+  if (!template) {
+    const templatePathWithDir = path.join('diff-nested-stacks-templates', templatePath);
+    template = JSON.parse(fs.readFileSync(path.join(__dirname, templatePathWithDir)).toString());
+    fs.writeFileSync(path.join(outdir, templatePath), JSON.stringify(template, undefined, 2));
+  }
+
+  for (const logicalId in template.Resources) {
+    if (template.Resources[logicalId].Type === 'AWS::CloudFormation::Stack') {
+      if (template.Resources[logicalId].Metadata && template.Resources[logicalId].Metadata['aws:asset:path']) {
+        const nestedTemplatePath = template.Resources[logicalId].Metadata['aws:asset:path'];
+        addNestedStacks(nestedTemplatePath, outdir);
+      }
+    }
+  }
+}
+
 function addAttributes(assembly: TestAssembly, builder: cxapi.CloudAssemblyBuilder) {
```

Comment:
I can't find where is this function called...? Is this change needed?

Reply:
it's called in [testAssembly()](https://github.com/aws/aws-cdk/blob/554645d1ca4d4223a0ab9490ea5cf5c0f201ed3a/packages/aws-cdk/test/util.ts#L87-L100). The new `addNestedStacks()` adds the nested template files to the testing environment, because the test files are all copied to a unique folder and if we don't copy these templates then they can't be found.